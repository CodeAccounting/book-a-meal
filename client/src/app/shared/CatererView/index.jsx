import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Header from '../../shared/Header';
import Footer from '../../shared/Footer';
import SideNav from '../../shared/SideNav';
import Preloader from '../../shared/Preloader';
import { userPropTypes, } from '../../../helpers/proptypes';

/**
 * @exports
 * @function CatererView
 * @desc Creates CatererView Component
 * @returns {JSX} CatererView Component
 */
const CatererView = ({
  user, logout, type, isFetching, children, showTime
}) => {
  const mainClass = classNames({
    'content-wrapper': true,
    dashboard: type === 'dashboard'
  });

  return (
    <div className="admin">
      <Header type="caterer" dateType={type} showTime={showTime} />
      <div className="content">
        <SideNav user={user} logout={logout} active={type} />
        <div className={mainClass} id="has-modal">
          {isFetching && <Preloader />}
          {!isFetching && children}
        </div>
      </div>
      <Footer />
    </div>
  );
};

CatererView.propTypes = {
  ...userPropTypes,
  isFetching: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  showTime: PropTypes.bool.isRequired
};

export default CatererView;
