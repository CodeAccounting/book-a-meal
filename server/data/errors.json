{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}