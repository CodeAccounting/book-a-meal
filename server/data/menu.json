[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]