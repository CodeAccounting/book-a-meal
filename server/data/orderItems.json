[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]