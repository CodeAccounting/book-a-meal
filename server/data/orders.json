[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]