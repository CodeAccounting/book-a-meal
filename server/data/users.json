[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
    "username": null,
    "businessName": "FoodCircle",
    "email": "food@circle.com",
    "password": "foodcircle",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  },
  {
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "username": "Iveren",
    "businessName": null,
    "email": "iveren@shaguy.com",
    "password": "iverenshaguy",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": null,
    "businessAddress": null,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "user"
  },
  {
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "username": "Olisa",
     "businessName": null,
     "email": "olisa@emodi.com",
     "password": "olisaemodi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "username": "Emiola",
     "businessName": null,
     "email": "emiola@olasanmi.com",
     "password": "emiolaolasanmi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
    "username": null,
    "businessName": "BellyFill",
    "email": "belly@fill.com",
    "password": "bellyfil",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  }
]