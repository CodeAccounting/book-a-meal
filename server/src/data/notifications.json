[
  {
    "notifId": "3dc9cc9b-179b-4f0d-ac60-7ca6f4659d79",
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": true,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "87a311b7-898b-4a6b-81c2-619417c96432",
    "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": true,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "9015c150-6079-4be6-a240-272a4de6bd17",
    "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": false,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "38eb0942-38b7-4a16-91e9-8b1662175d75",
    "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": false,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "d886ce22-9b41-4cf6-9dd6-7bdd5476d6c0",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
     "isRead": false,
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]