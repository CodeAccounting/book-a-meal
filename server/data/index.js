import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
