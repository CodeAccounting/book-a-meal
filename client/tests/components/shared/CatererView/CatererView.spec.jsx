import React from 'react';
import CatererView from '../../../../src/components/shared/CatererView';
import { caterer } from '../../../setup/data';

const props = {
  user: caterer,
  logout: jest.fn(),
  type: 'dashboard',
  isFetching: false,
  children: <p>Hi</p>,
  showTime: true
};
const { now } = Date;

describe('CatererView', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly', () => {
    const wrapper = shallow(<CatererView {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p').length).toEqual(1);
    expect(wrapper.find('Preloader').length).toBeFalsy();
  });

  it('renders Preloader when fetching', () => {
    const wrapper = shallow(<CatererView {...props} isFetching />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Preloader').length).toEqual(1);
    expect(wrapper.find('p').length).toBeFalsy();
  });
});
