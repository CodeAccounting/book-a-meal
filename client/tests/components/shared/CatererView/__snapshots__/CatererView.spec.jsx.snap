// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererView renders Preloader when fetching 1`] = `
<div
  className="admin"
>
  <Header
    dateType="dashboard"
    logout={[MockFunction]}
    showTime={true}
    type="caterer"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  />
  <div
    className="content"
  >
    <SideNav
      active="dashboard"
      logout={[MockFunction]}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    />
    <div
      className="content-wrapper dashboard"
      id="has-modal"
    >
      <Preloader
        type={null}
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`CatererView renders correctly 1`] = `
<div
  className="admin"
>
  <Header
    dateType="dashboard"
    logout={[MockFunction]}
    showTime={true}
    type="caterer"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  />
  <div
    className="content"
  >
    <SideNav
      active="dashboard"
      logout={[MockFunction]}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    />
    <div
      className="content-wrapper dashboard"
      id="has-modal"
    >
      <p>
        Hi
      </p>
    </div>
  </div>
  <Footer />
</div>
`;
